tests/e2e/container_image_pull_test.go

```diff
@@ -76,38 +108,86 @@ func pullAndVerifyQuayCDN(imageRef string) {
 	time.Sleep(1 * time.Second)
 
 	By("Verifying CDN requests in squid logs")
-	// Check logs from all pods since the beginning
-	cacheHitFound := false
+	// Collect logs from all pods and check for MISS and HIT patterns
+	var foundMiss, foundHit bool
+	missPattern := strings.Replace(cdnRegexPattern, "TCP_(MISS|HIT)", "TCP_MISS", 1)
+	hitPattern := strings.Replace(cdnRegexPattern, "TCP_(MISS|HIT)", "TCP_HIT", 1)
```

Comment:
This is quite brittle as it relies on the function caller to ensure `TCP_(MISS|HIT)` exists in the regex. If they don't provide it, unexpected behavior can ensue.

I think it'd be better to only accept a regex containing the host portion of the regex (e.g. `(docker-images-prod\.[a-f0-9]{32}\.r2\.cloudflarestorage\.com|production\.cloudflare\.docker\.com|docker-images-prod\.s3[a-z0-9.-]*\.amazonaws\.com)`) and then prefix that with `(?m)^.*TCP_MISS` or `(?m)^.*TCP_HIT` here.